Readme, startup, and Test Env Updates (#163)
AmplicaLabs/reconnection-service: README.md

# Graph Reconnection Service

<!-- TABLE OF CONTENTS -->

# 📗 Table of Contents

- [📖 About the Project](#about-project)
- [🔍 Arch Map](#-arch-maps)
- [🛠 Built With](#-built-with)
  - [Tech Stack](#tech-stack)
  - [Key Features](#key-features)
- [🚀 Live OpenAPI Docs](#-live-docs)
- [💻 Getting Started](#-getting-started)
  - [Prerequisites](#prerequisites)
  - [Setup](#setup)
  - [Install](#install)
  - [Usage](#usage)
  - [Run tests](#run-tests)
  - [Deployment](#deployment)
- [🤝 Contributing](#-contributing)
- [⭐️ Show your support](#-support)
- [🙏 Acknowledgements](#-acknowledgements)
- [❓FAQ](#faq)
- [📝 License](#-license)

<!-- PROJECT DESCRIPTION -->

> # 📖 [project_name] <a name="about-project"></a>

Reviewer: Is [project_name] supposed to render `Reconnection Service`?
```suggestion
# 📖 Reconnection Service <a name="about-project"></a>
```

> > Describe your project in 2 paragraphs.

Reviewer: ```suggestion

```

<p align="right">(<a href="#-table-of-contents">back to top</a>)</p>

<!-- GETTING STARTED -->

## 💻 Getting Started

To get a local copy up and running, follow these steps.

### Prerequisites

In order to run this project you need:

- [Node.js](https://nodejs.org)
- [Docker](https://docs.docker.com/get-docker/)

### Setup

Clone this repository to your desired folder:

Example commands:

```sh
  git clone git@github.com:AmplicaLabs/account-service.git
  cd account-service
```

### Environment Variables

Modify any environment variables in the `.env` file as needed. For docker compose env `.env.docker.dev` file is used. The complete set of environment variables is documented [here](./ENVIRONMENT.md), and a sample environment file is provided [here](./env.template).

1. Copy the template values into the .env files.
   ```sh
   cp env.template .env
   cp env.template .env.docker.dev
   ```
2. Replace template values with values appropriate to your environment.

### Install

Install NPM Dependencies:

```sh
  npm install
```

### Usage

Note: using [docker compose file](docker-compose.yaml) to start the services. This will start the services in development mode.

> In order to run the `account-service` in development mode without containers, you can use the following commands:

Reviewer: ```suggestion
In order to run the `reconnection-service` in development mode without containers, you can use the following commands:
```

> #### 1. Start the Redis server container, the Frequency container, and the mock webhook server. You can view the logs with your Docker setup.

Reviewer: ```suggestion
#### 1. Start the Redis server container and the Frequency container. You can view the logs with your Docker setup.
```

   ```bash
   docker compose up -d redis frequency
   ```

#### 2. Once [Frequency](https://polkadot.js.org/apps/?rpc=ws%3A%2F%2F127.0.0.1%3A9944#/explorer) is up, run the chain setup. This will setup 7000 MSAs on the chain with Graph.

   ```bash
   npm run chain-setup
   ```

   Note: If you change the env's `PROVIDER_ACCOUNT_SEED_PHRASE`, you will also need to update `PROVIDER_ACCOUNT_SEED_PHRASE` in `setup/testing/index.ts`.

#### 3. Start the mock webhook server. You can view the logs with your Docker setup.

  ```bash
  docker compose up -d webhook
  ```

#### 4. Start the Api.<br /><br />

   **Option 1:** In a new terminal window, start the `reconnection-service` app. Logs will be displayed in the terminal for easy reference.

   ```sh
   npm run start:api:debug
   ```

   -- or -- <br /><br />

   **Option 2:**

>    Run the following command to start the account service api and worker containers. This will start the account service api and worker in development mode.

Reviewer: ```suggestion
   Run the following command to start the reconnection service container. This will start the reconnection service in development mode.
```

   ```sh
   docker compose up -d reconnection-service
   ```

#### 5. Check the job in [BullUI](http://0.0.0.0:3000/reconnection-service/queue/), to monitor job progress based on defined tests.

## 📋 Testing

### Run the tests.

   ```bash
   npm run test
   ```

   This will run the matching `*.spec.ts` files.

#### Check e2e test file for more details on the test.

### Queue Management

You may also view and manage the application's queue at [http://localhost:3000/reconnection-service/queue](http://localhost:3000/reconnection-service/queue).

### Linting

```sh
  npm run lint
```

### Auto-format

```sh
  npm run format
```

### Debugging

> - Docker to stop containers, networks, volumes, and images created by `docker compose up` run...

Reviewer: ```suggestion
- Docker: to stop containers, networks, volumes, and images created by `docker compose up` run...
```
